tools/swarm/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+        Self::PathConversion(error)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::Swarm;
+
+    const IMAGE: &str = "hyperledger/iroha:dev";
+    const PEER_CONFIG_PATH: &str = "./configs/swarm";
+    const TARGET_PATH: &str = "./configs/swarm/docker-compose.yml";
+
+    // `serde_yaml` doesn't format YAML well – it's better to compare structurally.
+    // Otherwise sample files will break tests on reformats.
+    fn assert_yml_eq(expected: &str, actual: &str) {
+        let expected: serde_yaml::Value =
+            serde_yaml::from_str(expected).expect("should deserialize expected");
+        let actual: serde_yaml::Value =
+            serde_yaml::from_str(actual).expect("should deserialize actual");
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn single_build_banner() {
+        const EXPECTED: &str = include_str!("../samples/single-build-banner.yml");
+        let mut buffer = Vec::new();
+        Swarm::new(
+            nonzero_ext::nonzero!(1u16),
+            Some(&[]),
+            false,
+            PEER_CONFIG_PATH.as_ref(),
+            IMAGE,
+            Some(".".as_ref()),
+            false,
+            TARGET_PATH.as_ref(),
+        )
+        .unwrap()
+        .build()
+        .write(&mut buffer, Some(&["Single-line banner"]))
+        .unwrap();
+        let actual = std::str::from_utf8(&buffer).unwrap();
+        assert_yml_eq(EXPECTED, actual);
+    }
+
+    #[test]
+    fn single_build_banner_nocache() {
+        const EXPECTED: &str = include_str!("../samples/single-build-banner-nocache.yml");
+        let mut buffer = Vec::new();
+        Swarm::new(
+            nonzero_ext::nonzero!(1u16),
+            Some(&[]),
+            false,
+            PEER_CONFIG_PATH.as_ref(),
+            IMAGE,
+            Some(".".as_ref()),
+            true,
+            TARGET_PATH.as_ref(),
+        )
+        .unwrap()
+        .build()
+        .write(
+            &mut buffer,
+            Some(&["Multi-line banner 1", "Multi-line banner 2"]),
+        )
+        .unwrap();
+        let actual = std::str::from_utf8(&buffer).unwrap();
+        assert_yml_eq(EXPECTED, actual);
+    }
+
+    #[test]
+    fn single_pull_healthcheck() {
+        const EXPECTED: &str = include_str!("../samples/single-pull-healthcheck.yml");
+        let mut buffer = Vec::new();
+        Swarm::new(
+            nonzero_ext::nonzero!(1u16),
+            Some(&[]),
+            true,
+            PEER_CONFIG_PATH.as_ref(),
+            IMAGE,
+            None,
+            false,
+            TARGET_PATH.as_ref(),
+        )
+        .unwrap()
+        .build()
+        .write(&mut buffer, None)
+        .unwrap();
+        let actual = std::str::from_utf8(&buffer).unwrap();
+        assert_yml_eq(EXPECTED, actual);
+    }
+
+    #[test]
+    fn multiple_pull_healthcheck_nocache() {
+        const EXPECTED: &str = include_str!("../samples/multiple-pull-healthcheck-nocache.yml");
```

Comment:
Having separate `yml` files with fixtures is convenient from files structuring point, but it forces to manually update them whenever they are updated. The reason I used `expect_test` is that it can rewrite the snapshots automatically with `UPDATE_EXPECT=1` env variable.

Actally, just discovered existence of [`expect_test::expect_file!`](https://docs.rs/expect-test/latest/expect_test/macro.expect_file.html) macro. I would prefer if you use it.

Reply:
Going back to using `expect!` because `expect_file!` doesn't support custom asserts, and the files will break on formatting due to the way `serde_yaml` serializes lists.

Reply:
Ah, okay :(